web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/AutoDiscoverToggle.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import { Box, Toggle } from 'design';
+
+import { ToolTipInfo } from 'shared/components/ToolTip';
+
+export function AutoDiscoverToggle({
+  wantAutoDiscover,
+  toggleWantAutoDiscover,
+  disabled = false,
+}: {
+  wantAutoDiscover: boolean;
+  toggleWantAutoDiscover(): void;
+  disabled?: boolean;
+}) {
+  return (
+    <Box mb={2}>
+      <Toggle
+        isToggled={wantAutoDiscover}
+        onToggle={toggleWantAutoDiscover}
+        disabled={disabled}
+      >
```

Comment:
There's a lot of flashing when this toggle is clicked. For example, I shouldn't see "No results" for a split second when switching between toggle states. Presumably it's because of those extra effects used in `AutoEnrollment` and `SingleEnrollment` that I mentioned. Maybe addressing them will make the UI feel more stable as well.

Reply:
i made it less jumpy, but couldn't resolve why `no results` keep showing up before the actual data, it's strange, these are the render ticks. the table for some reason gets an empty item first, then the data comes on next render...

it's like `setDataTable` gets partially updated (because the `fetchStatus: disabled` is updated but not the items)...

<img width="670" alt="image" src="https://github.com/user-attachments/assets/3dc6d116-c60e-4efb-8820-518584116ec3">